fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java

```diff
@@ -1540,6 +1529,68 @@ public void modifyJob(RoutineLoadDesc routineLoadDesc,
         }
     }
 
+    public void mergeLoadDescToOriginStatement(RoutineLoadDesc routineLoadDesc) {
+        if (origStmt == null) {
+            return;
+        }
+
+        RoutineLoadDesc originLoadDesc = CreateRoutineLoadStmt.getLoadDesc(origStmt, sessionVariables);
+        if (originLoadDesc == null) {
+            originLoadDesc = new RoutineLoadDesc();
+        }
+        if (routineLoadDesc.getColumnSeparator() != null) {
+            originLoadDesc.setColumnSeparator(routineLoadDesc.getColumnSeparator());
+        }
+        if (routineLoadDesc.getRowDelimiter() != null) {
+            originLoadDesc.setRowDelimiter(routineLoadDesc.getRowDelimiter());
+        }
+        if (routineLoadDesc.getColumnsInfo() != null) {
+            originLoadDesc.setColumnsInfo(routineLoadDesc.getColumnsInfo());
+        }
+        if (routineLoadDesc.getWherePredicate() != null) {
+            originLoadDesc.setWherePredicate(routineLoadDesc.getWherePredicate());
+        }
+        if (routineLoadDesc.getPartitionNames() != null) {
+            originLoadDesc.setPartitionNames(routineLoadDesc.getPartitionNames());
+        }
+
+        // we use sql to persist the load properties, so we just put the load properties to sql.
+        String sql = "CREATE ROUTINE LOAD job ON tbl "
+                + originLoadDesc.toSql()
+                + " PROPERTIES (\"desired_concurrent_number\"=\"1\")"
+                + " FROM KAFKA (\"kafka_topic\" = \"my_topic\")";
+        LOG.debug("merge result: {}", sql);
+        origStmt = new OriginStatement(sql, 0);
+    }
+
+    public RoutineLoadDesc getLoadDesc(OriginStatement origStmt, Map<String, String> sessionVariables)
```

Comment:
This function seems to be identical with CreateRoutineLoadStmt.getLoadDesc()? Is it useful? I can't find the usage of this one.

Reply:
I forget to remove it.

Reply:
done